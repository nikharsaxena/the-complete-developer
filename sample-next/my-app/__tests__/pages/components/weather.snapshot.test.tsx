/**
 * @jest-environment node
 */

import PageComponentWeather from "../../../pages/components/weather";
import { act, create } from "react-test-renderer";

describe("PageComponentWeather", () => {
    test("renders correctly", async () => {
        let component: any;
        await act(async () => {
            component = await create(<PageComponentWeather></PageComponentWeather>);
        })
        expect(component.toJSON()).toMatchSnapshot();
    });

    test("clicks the h1 element and updates the state", async () => {
        let component: any;
        await act(async () => {
            component = await create(<PageComponentWeather></PageComponentWeather>);
            component.root.findByType("h1").props.onClick();
        });
        expect(component.toJSON()).toMatchSnapshot();
    });
})