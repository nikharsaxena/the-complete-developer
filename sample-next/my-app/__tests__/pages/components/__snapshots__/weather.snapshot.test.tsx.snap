// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageComponentWeather clicks the h1 element and updates the state 1`] = `
<h1
  onClick={[Function]}
>
  The weather is 
  sunny
  , and the counter shows 
  1
</h1>
`;

exports[`PageComponentWeather renders correctly 1`] = `
<h1
  onClick={[Function]}
>
  The weather is 
  sunny
  , and the counter shows 
  0
</h1>
`;
